spinalcordtoolbox/resampling.py

```diff
@@ -56,6 +56,14 @@ def resample_nib(image, new_size=None, new_size_type=None, image_dest=None, inte
     else:
         raise TypeError(f'Invalid image type: {type(image)}')
 
+    # convert to floating point if we're doing arithmetic interpolation
+    if interpolation != 'nn' and img.get_data_dtype().kind in 'biu':
+        original_dtype = img.get_data_dtype()
+        img = nib.nifti1.Nifti1Image(img.get_fdata(), img.affine, img.header)
```

Comment:
I think we may need to use `img.header.get_best_affine()` rather than `img.affine`? There have been some code issues in the past related to this: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3295#issuecomment-833966224

> Assuming that `img.affine == img.header.get_best_affine()` (which will be true if you're loading `img` from file), then `Nifti1Image(data, img.affine, img.header)` will get you the same thing, but also allow you go modify the file. If you don't trust your affine but do trust your header, then `Nifti1Image(data, img.header.get_best_affine(), img.header)` should work fine.

Since this change is happening inside `resample_nib`, rather than `resample_file`, I'm not sure if we can safely assume that `img.affine` produces the same result as `img.hdr.get_best_affine()`? So, using the latter I think should help us preserve the codes upon initializing a new image.

Reply:
Thanks for the suggestions, you're entirely right. I think I was relying on the call to `img.header.get_best_affine()` a few lines above, but of course that only covers the case of `isinstance(image, Image)`, not the case of `isinstance(image, nib.nifti1.Nifti1Image)`.
```suggestion
        img = nib.nifti1.Nifti1Image(img.get_fdata(), img.header.get_best_affine(), img.header)
```